src/state/context/UserContext.jsx

```diff
@@ -0,0 +1,66 @@
+import { createContext, useEffect, useMemo, useState } from 'react';
+import {
+  getUser,
+  getLocalProfile,
+  saveLocalProfile,
+  removeLocalProfile,
+  getProfile,
+  onAuthChange,
+} from '../services/user-service.js';
+
+export const UserStateContext = createContext();
+export const UserActionContext = createContext();
+
+// eslint-disable-next-line react/prop-types
+export default function UserProvider({ children }) {
+  const [user, setUser] = useState(getUser());
+  const [profile, setProfile] = useState(getLocalProfile());
+
+  const loadProfile = async () => {
+    const { data, error } = await getProfile();
+    if (error) {
+      console.log(error);
```

Comment:
don't forget to clean up your code by taking out console logs when you are ready for a code review